Move cruntime to separate module

noahemmet opened the pull request:
This moves `cruntime` to a separate module, which allows this framework to be built on Linux. Discussion: https://github.com/wickwirew/Runtime/issues/18#issuecomment-434533969 and https://github.com/noahemmet/Runtime/commit/8d685eb4ff7be23385bac54c320aaa71117b38e3

- It looked like the last few commits included `*.xcodeproj` and `*.xcworkspace` files, but if you'd like, I'm happy to add them to the `.gitignore`.
- Tests pass locally on my Mac but I'm not set it up test on Linux; is that something Travis will do automatically?

Thanks for everyone's help on this!

noahemmet commented on Runtime/Utilities/GetFieldAt.swift:
```diff
@@ -42,8 +34,12 @@ func getProperties(of type: Any.Type, offsets: [Int]) -> [PropertyInfo] {
-            let infoContext = ctx.assumingMemoryBound(to: PropertyInfoContext.self).mutable
+        let pointer = unsafeBitCast(type, to: UnsafeRawPointer.self)
+        swift_getFieldAt(pointer, UInt32(index), { name, type, ctx in
+            guard let name = name, let ctx = ctx else { 
```
Does this look correct? Should this be a `fatalError` instead perhaps? I'm not sure under what circumstances these would be `nil`.

wickwirew replied:
Yea I would do a `fatalError` instead. It shouldn't ever be `nil` and if it is it will fail somewhere else because of it.

noahemmet commented:
I was getting a `error: /Users/travis/build/wickwirew/Runtime/cruntime has no manifest` in [Travis](https://travis-ci.org/wickwirew/Runtime/jobs/457234678) so I added a `Package.swift` in the `cruntime` directory. The MacOS tests now pass but Linux is [giving me](https://travis-ci.org/wickwirew/Runtime/jobs/457244157) a `error: cannot find -lcruntime`.

wickwirew commented:
Not entirely sure. I don't have much experience with SwiftPM. My guess is the `Runtime` package needs the `CRuntime` package as a dependency

wickwirew commented:
It doesn't look like you can have 2 packages in the same repo as explained [here](https://stackoverflow.com/questions/48666237/swift-package-manager-dependency-in-a-repository-subdirectory). I can create a new repo for the package if need be

noahemmet commented:
Ah would you mind? I can PR `cruntime` against it. 

PS, would you prefer this be named `cruntime` or `CRuntime`? I wasn't sure about prior art here.

wickwirew commented:
Would prefer `CRuntime` just to keep it consistent if you wouldn't mind updating it. And created it https://github.com/wickwirew/CRuntime

noahemmet commented:
Hey, tests pass! I've updated the dependency to point to `wickwirew/CRuntime.git`, so feel free to re-run them after https://github.com/wickwirew/CRuntime/pull/1 is merged.

A few things:
- I moved `Runtime` to `Sources/Runtime` (ditto for `Tests`). It was helpful for debugging SPM, but I don't think it's strictly necessary for this PR; I'm happy to revert it if you'd like.
- The latest SPM did some nice test autogeneration (see https://github.com/wickwirew/Runtime/pull/31/files#diff-b7f3e3662900e97b99353ab438d29031), so I left that in there.

Thanks again to everyone who helped!

noahemmet commented:
I've verified this works by pushing a GraphQL app to Vapor's Cloud service here: https://graphql-star-wars.v2.vapor.cloud/graphiql

wickwirew commented:
Sorry for the late response! Slipped my mind. Thanks again for tackling this!
